src/plots/cartesian/axes.js

```diff
@@ -2636,6 +2638,39 @@ axes.drawOne = function(gd, ax, opts) {
     return Lib.syncOrAsync(seq);
 };
 
+function keepSelectedAutoMargin(automargin, push) {
+    if(typeof automargin === 'boolean') return push;
+
+    var keepMargin = [];
+    var mapping = {
```

Comment:
It looks like `mapping` is a constant. 
Let's use all caps name and move it outside the scope of this function to avoid creation of the object every time calling the function. 
Thanks.

Reply:
fixed